jetty-core/jetty-quic/jetty-quic-api/src/main/java/org/eclipse/jetty/quic/api/Stream.java

```diff
@@ -0,0 +1,376 @@
+     * <p>A stream becomes remotely closed when it has {@link #read()}
+     * the last data.</p>
+     *
+     * @return whether the stream is remotely closed
+     * @see #isLocallyClosed()
+     * @see #isClosed()
+     */
+    boolean isRemotelyClosed();
+
+    long getIdleTimeout();
+
+    void setIdleTimeout(long idleTimeout);
+
+    /**
+     * <p>Returns the {@link Session} this stream is associated to.</p>
+     *
+     * @return the {@link Session} this stream is associated to
+     */
+    Session getSession();
+
+    /**
+     * <p>Reads data from this stream, if any.</p>
+     * <p>The returned {@link Content.Chunk} object may be {@code null} indicating
+     * that the end of the read side of the stream has not yet been reached,
+     * which may happen when not all the bytes have been received so far,
+     * for example the remote peer did not send them yet, or they are
+     * in-flight.</p>
+     * <p>When the returned {@link Content.Chunk} object is not {@code null},
+     * the flow control window has been enlarged by the data length.
+     * Applications <em>must</em> call, either immediately or later (even
+     * asynchronously from a different thread) {@link Content.Chunk#release()}
+     * to notify the implementation that the bytes have been processed.</p>
+     * <p>{@link Content.Chunk} objects may be stored away for later,
+     * asynchronous, processing (for example, to process them only when
+     * all of them have been received).</p>
+     * <p>Once the returned {@link Content.Chunk} object indicates that the end
+     * of the read side of the stream has been reached, further calls to this
+     * method will return a {@link Content.Chunk} object with the same indication,
+     * although the instance may be different.</p>
+     *
+     * @return a {@link Content.Chunk} object containing the data bytes
+     * or a failure, or {@code null} if no data bytes are available
+     * @see #demand()
+     */
+    Content.Chunk read();
+
+    /**
+     * <p>Demands more data bytes for this stream.</p>
+     * <p>Calling this method causes {@link Listener#onDataAvailable(Stream)}
+     * to be invoked, possibly at a later time, when the stream has data
+     * to be read, but also when the stream has reached EOF.</p>
+     * <p>This method is idempotent: calling it when there already is an
+     * outstanding demand to invoke {@link Listener#onDataAvailable(Stream)}
+     * is a no-operation.</p>
+     * <p>The thread invoking this method may invoke directly
+     * {@link Listener#onDataAvailable(Stream)}, unless another thread
+     * that must invoke {@link Listener#onDataAvailable(Stream)}
+     * notices the outstanding demand first.</p>
+     * <p>It is always guaranteed that invoking this method from within
+     * {@code onDataAvailable(Stream)} will not cause a
+     * {@link StackOverflowError}.</p>
+     *
+     * @see #read()
+     * @see Listener#onDataAvailable(Stream)
+     */
+    void demand();
```

Comment:
There is nothing in this abstract contract that allows a Listener to be associated with a Stream?

Reply:
Hmmm now I notice that `HttpStream` is-not-a `Content.Source`.... hmmmm